File: online-judge-solutions/Codeforces/609A.py
#https://codeforces.com/problemset/problem/609/A
#Lang		:	Python 3.8
#Time		:	46 ms
#Memory         :	0 KB
n,m=int(input()),int(input())
s=[0]*(n)
a=sorted([int(input()) for i in range(n)],reverse=True)
s[0]=a[0]
for i in range(1,n,1):s[i]=s[i-1]+a[i]
print(next((i for i in range(n) if s[i]>=m),None)+1)

File: online-judge-solutions/Codeforces/609A.cpp
//https://codeforces.com/problemset/problem/609/A
//Lang		:	GNU C++14
//Time		:	15 ms
//Memory	:	0 KB
#include<iostream>
#include<algorithm>
#include<vector>
using namespace std;
int main(){
	int n,m,r=0;cin>>n>>m;
	vector<int> a(n),s(n);
	for(int i=0;i<n;i++) cin>>a[i];
	sort(rbegin(a),rend(a));
	s[0]=a[0];
	for(int i=1;i<n;i++) s[i]=s[i-1]+a[i];
	for(int i=0;i<n;i++) if(s[i]>=m){r=i;break;}
	cout<<r+1<<endl;
	return 0;
}
